File: app/src/main/java/com/mobile/hinde/spacetime/Frag_ImageList.java
package com.mobile.hinde.spacetime;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Color;
import android.graphics.Typeface;
import android.net.Uri;
import android.os.AsyncTask;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.support.v4.content.res.ResourcesCompat;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.QueryDocumentSnapshot;
import com.google.firebase.firestore.QuerySnapshot;
import com.google.firebase.storage.FirebaseStorage;
import com.google.firebase.storage.StorageReference;
import com.mobile.hinde.utils.Grid_Adapter;
import com.mobile.hinde.utils.UserSettings;

import org.json.JSONException;

import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class Frag_ImageList extends Fragment {

    private final FirebaseStorage mStorage = FirebaseStorage.getInstance();
    private final FirebaseFirestore dbInstance = FirebaseFirestore.getInstance();

    private final UserSettings instance = UserSettings.getInstance();

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.frag_image_list,
                container, false);

        dbInstance.collection("users").document(instance.getUserId()).get().addOnCompleteListener(new OnCompleteListener<DocumentSnapshot>() {
            @Override
            public void onComplete(@NonNull Task<DocumentSnapshot> task) {
                if (task.isSuccessful()) {
                    DocumentSnapshot document = task.getResult();
                    Map<String,Object> userMap = document.getData();

                    // Create LinearLayout
                    final LinearLayout ll = new LinearLayout(getActivity());
                    ll.setOrientation(LinearLayout.VERTICAL);
                    TextView title = createTextView();
                    View line = createLine();



                    RecyclerView rv = createRecycler((ArrayList<String>) userMap.get("MOON"));

                    LinearLayout root = getView().findViewById(R.id.rootLayer);

                    ll.addView(title);
                    ll.addView(line);
                    ll.addView(rv);

                    root.addView(ll);

                } else {
                }
            }
        });
        return view;
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    public void onDestroy(){
        super.onDestroy();
    }

    private TextView createTextView(){
        TextView txt = new TextView(getActivity());

        Typeface font = ResourcesCompat.getFont(getActivity(), R.font.digitaldream);
        txt.setTypeface(font);
        txt.setTextSize(20);
        txt.setText("MOON");
        txt.setTextColor(ContextCompat.getColor(getActivity(),R.color.colorPrimary));
        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT,LinearLayout.LayoutParams.WRAP_CONTENT);
        txt.setLayoutParams(params);

        return txt;
    }

    private View createLine(){
        View v = new View(getActivity());
        ViewGroup.LayoutParams params = new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,5);
        v .setLayoutParams(params);
        v.setBackgroundColor(ContextCompat.getColor(getActivity(),R.color.colorPrimary));
        return v;
    }

    private RecyclerView createRecycler(ArrayList<String> imgList){
        final RecyclerView rv = new RecyclerView(getActivity());
        RecyclerView.LayoutParams params = new RecyclerView.LayoutParams(RecyclerView.LayoutParams.MATCH_PARENT,RecyclerView.LayoutParams.MATCH_PARENT);
        rv.setLayoutParams(params);

        Grid_Adapter customAdapter = new Grid_Adapter(getActivity(), imgList);
        rv.setAdapter(customAdapter); // set the Adapter to RecyclerView
        // set a GridLayoutManager with 3 number of columns , horizontal gravity and false value for reverseLayout to show the items from start to end
        GridLayoutManager gridLayoutManager = new GridLayoutManager(getActivity().getApplicationContext(),2, LinearLayoutManager.HORIZONTAL,false);
        rv.setLayoutManager(gridLayoutManager); // set LayoutManager to RecyclerView
        return rv;
    }
}

File: app/src/main/java/com/mobile/hinde/spacetime/FragImageList.kt
package com.mobile.hinde.spacetime

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.support.v4.content.res.ResourcesCompat
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.DisplayMetrics
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.ScrollView
import android.widget.TextView

import com.google.firebase.firestore.FirebaseFirestore
import com.mobile.hinde.utils.Constant
import com.mobile.hinde.utils.GridAdapter
import com.mobile.hinde.utils.UserSettings

import java.util.ArrayList

class FragImageList : Fragment() {

    private val dbInstance = FirebaseFirestore.getInstance()

    private var mNumColumn = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val dm = DisplayMetrics()
        activity!!.windowManager.defaultDisplay.getMetrics(dm)

        mNumColumn = (dm.widthPixels / TypedValue.applyDimension(
                TypedValue.COMPLEX_UNIT_DIP,
                120f,
                dm
        )).toInt()

    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.frag_image_list,
                container, false)

        dbInstance.collection("users").document(UserSettings.userId).get().addOnCompleteListener { task ->
            if (task.isSuccessful) {
                val document = task.result
                val userMap = document!!.data

                val root = getView()!!.findViewById<ScrollView>(R.id.scrollLayout)

                // Create LinearLayout
                val ll = LinearLayout(activity)
                ll.orientation = LinearLayout.VERTICAL


                for (target in Constant.TARGET_LIST) {
                    if (userMap!!.containsKey(target) && (userMap[target] as ArrayList<String>).size > 0) {
                        val title = createTextView(target)
                        val line = createLine()

                        val rv = createRecycler(userMap[target] as ArrayList<String>)

                        ll.addView(title)
                        ll.addView(line)
                        ll.addView(rv)
                    }
                }

                root.addView(ll)

            } else {
            }
        }
        return view
    }

    private fun createTextView(target: String): TextView {
        val txt = TextView(activity)

        val font = ResourcesCompat.getFont(activity!!, R.font.digitaldream)
        txt.typeface = font
        txt.textSize = 20f
        txt.text = target
        txt.setTextColor(ContextCompat.getColor(activity!!, R.color.colorPrimary))
        val params = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT)
        params.setMargins(20, 50, 0, 20)
        txt.layoutParams = params

        return txt
    }

    private fun createLine(): View {
        val v = View(activity)
        val params = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, 5)
        v.layoutParams = params
        v.setBackgroundColor(ContextCompat.getColor(activity!!, R.color.colorPrimary))
        return v
    }

    private fun createRecycler(imgList: ArrayList<String>): RecyclerView {
        val rv = RecyclerView(activity!!)
        val params = RecyclerView.LayoutParams(RecyclerView.LayoutParams.MATCH_PARENT, RecyclerView.LayoutParams.WRAP_CONTENT)
        params.setMargins(20, 20, 20, 20)
        rv.layoutParams = params

        val customAdapter = GridAdapter(activity!!, imgList)
        rv.adapter = customAdapter // set the Adapter to RecyclerView
        // set a GridLayoutManager with 3 number of columns , horizontal gravity and false value for reverseLayout to show the items from start to end

        val gridLayoutManager = GridLayoutManager(activity!!.applicationContext, mNumColumn, LinearLayoutManager.VERTICAL, false)
        rv.layoutManager = gridLayoutManager // set LayoutManager to RecyclerView
        return rv
    }
}
